### PR TITLE
Update exception handling in the metadata watcher.

### PR DESCRIPTION
**Pull request by illfelder**
* An error message should only be printed once to prevent log spam.
* Calls to the metadata watcher will retry until metadata is retrieved.

### CODE REVIEW

File: google_compute_engine/metadata_watcher.py [hunk 1]
```diff
@@ -140,6 +140,30 @@ def _GetMetadataUpdate(self, metadata_key='', recursive=True, wait=True):
...
+      try:
+        return self._GetMetadataUpdate(
+            metadata_key=metadata_key, recursive=recursive, wait=wait)
+      except (httpclient.HTTPException, socket.error, urlerror.URLError) as e:
+        if isinstance(e, type(exception)) and e.args == exception.args:
+          continue
+        else:
+          exception = e
+          message = 'GET request error retrieving metadata. %s.'
+          self.logger.error(message, str(exception))
```
Comment (jonparrott) [inline google_compute_engine/metadata_watcher.py]: why `str(exception)`, why not just `exc_info=True`?

Comment (jonparrott) [inline google_compute_engine/metadata_watcher.py]: Or even better, since you're doing `.error` just do `.exception`.

Comment (illfelder) [inline google_compute_engine/metadata_watcher.py]: Thanks!

**Comment by zmarano** [comment]
LGTM

**Pull request merged**